libs/application/templates/aosh/register-new-machine/src/lib/RegisterNewMachineTemplate.ts

```diff
@@ -0,0 +1,228 @@
+                {
+                  event: DefaultEvents.SUBMIT,
+                  name: 'Staðfesta',
+                  type: 'primary',
+                },
+              ],
+              write: 'all',
+              read: 'all',
+              delete: true,
+              api: [
+                IdentityApi,
+                UserProfileApi,
+                MachineTypesApi,
+                MachineParentCategoriesApi,
+              ],
+            },
+          ],
+        },
+        on: {
+          [DefaultEvents.SUBMIT]: { target: States.DRAFT },
+        },
+      },
+      [States.DRAFT]: {
+        meta: {
+          name: 'Tilkynning um eigendaskipti að tæki',
+          status: 'draft',
+          actionCard: {
+            tag: {
+              label: applicationMessage.actionCardDraft,
+              variant: 'blue',
+            },
+            historyLogs: [
+              {
+                logMessage: coreHistoryMessages.applicationSent,
+                onEvent: DefaultEvents.SUBMIT,
+              },
+            ],
+          },
+          lifecycle: EphemeralStateLifeCycle,
+
+          roles: [
+            {
+              id: Roles.APPLICANT,
+              formLoader: () =>
+                import('../forms/RegisterMachineForm/index').then((module) =>
+                  Promise.resolve(module.RegisterNewMachineForm),
+                ),
+              actions: [
+                {
+                  event: DefaultEvents.SUBMIT,
+                  name: 'Staðfesta',
+                  type: 'primary',
+                },
+              ],
+              write: 'all',
+              delete: true,
+            },
+          ],
+        },
+        on: {
+          [DefaultEvents.SUBMIT]: { target: States.COMPLETED },
+        },
+      },
+      [States.COMPLETED]: {
+        meta: {
+          name: 'Completed',
+          status: 'completed',
+          lifecycle: pruneAfterDays(30),
+          onEntry: defineTemplateApi({
+            action: ApiActions.submitApplication,
+          }),
+          actionCard: {
+            tag: {
+              label: applicationMessage.actionCardDone,
+              variant: 'blueberry',
+            },
+            pendingAction: {
+              title: corePendingActionMessages.applicationReceivedTitle,
+              displayStatus: 'success',
+            },
+          },
+          roles: [
+            {
+              id: Roles.APPLICANT,
+              formLoader: () =>
+                import('../forms/Conclusion').then((module) =>
+                  Promise.resolve(module.Conclusion),
+                ),
+              read: 'all',
+            },
+          ],
+        },
+      },
+    },
+  },
+  stateMachineOptions: {
+    actions: {
+      assignUsers: assign((context) => {
+        const { application } = context
+
+        const buyerNationalId = getBuyerNationalId(application)
+        if (buyerNationalId !== null && buyerNationalId !== '') {
```

Comment:
What if buyerNationalId is undefined?
This could just be `if (!buyerNationalId)`

Reply:
I don't even know why this is there, there is no assignee. Will remove this.